code/defines/obj.dm

```diff
@@ -326,8 +326,9 @@ var/global/list/PDA_Manifest = list()
 	siemens_coefficient = 1
 
 /obj/item/weapon/beach_ball/afterattack(atom/target as mob|obj|turf|area, mob/user as mob)
-	if(user.drop_item(src))
-		src.throw_at(target, throw_range, throw_speed)
+	if(src.loc == user)
```

Comment:
From what I read of the issue this is better solved by making the attackby return 1 when loading is successful